src/python/pants/engine/internals/graph.py

```diff
@@ -321,13 +322,12 @@ async def transitive_targets(targets: Targets) -> TransitiveTargets:
 
     _detect_cycles(tuple(t.address for t in targets), dependency_mapping)
 
-    transitive_wrapped_excludes = await MultiGet(
-        Get(WrappedTarget, AddressInput, address_input)
+    nested_transitive_excludes = await MultiGet(
+        Get(Targets, UnparsedAddressInputs, t.get(Dependencies).unevaluated_transitive_excludes)
         for t in (*targets, *visited)
-        for address_input in t.get(Dependencies).unevaluated_transitive_excludes
     )
```

Comment:
This is less efficient than it could be. More efficient would be to only call `Get(Targets, UnparsedAddressInputs)` if there are actual values in it. It would avoid overhead with invoking rules, even though Pantsd reduces the cost.

But until we can use the walrus operator, this means using a for-loop to set up all the request objects, otherwise we'd have to call `t.get(Dependencies)` twice, which is non-trivial (it iterates over every field on the target looking for subclasses).

I'm not sure if it's worth the slightly less readable code to use a for loop. It probably is.

Reply:
I would do the right thing for performance here, even if it's mildly clunkier.